test-case/check-alsa-conformance.sh

```diff
@@ -0,0 +1,251 @@
+
+# Copyright(c) 2025 Intel Corporation.
+# SPDX-License-Identifier: BSD-3-Clause
+
+##
+## Case Name: Execute ALSA conformance tests.
+##
+## Preconditions:
+##    - ChromeOS Audio Test package is installed
+##      https://chromium.googlesource.com/chromiumos/platform/audiotest
+##
+## Description:
+##    Run `alsa_conformance_test.py` for the playback device
+##    and for the capture devices with the test suite paramenters given.
+##    Compose resulting JSON reports.
+##
+## Case steps:
+##    0. Set ALSA parameters.
+##    1. Try to start `alsa_conformance_test` in device info mode.
+##    2. Start `alsa conformance_test.py` for playback device.
+##    3. Start `alsa conformance_test.py` for capture device.
+##    4. Compose the resulting JSON report.
+##
+## Expect result:
+##    ALSA conformance results collected and saved in `test_result.json` file.
+##    Exit status 0.
+##
+
+TESTDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+TESTLIB="${TESTDIR}/case-lib"
+
+# shellcheck source=case-lib/lib.sh
+source "${TESTLIB}/lib.sh"
+
+OPT_NAME['d']='device'          OPT_DESC['d']='ALSA pcm device for playback and capture. Example: hw:0'
+OPT_HAS_ARG['d']=1              OPT_VAL['d']=''
+
+OPT_NAME['p']='pcm_p'           OPT_DESC['p']='ALSA pcm device for playback only. Example: hw:soundwire,0'
+OPT_HAS_ARG['p']=1              OPT_VAL['p']=''
+
+OPT_NAME['c']='pcm_c'           OPT_DESC['c']='ALSA pcm device for capture only. Example: hw:soundwire,1'
+OPT_HAS_ARG['c']=1              OPT_VAL['c']=''
+
+OPT_NAME['r']='rates'           OPT_DESC['r']='Sample ratis to try. Default: check all available rates.'
+OPT_HAS_ARG['r']=1              OPT_VAL['r']=''
+
+OPT_NAME['F']='formats'         OPT_DESC['F']='Data formats to try. Default: check all available formats.'
+OPT_HAS_ARG['F']=1              OPT_VAL['F']=''
+
+OPT_NAME['s']='sof-logger'      OPT_DESC['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_HAS_ARG['s']=0              OPT_VAL['s']=1
+
+OPT_NAME['v']='verbose'         OPT_DESC['v']='Verbose logging.'
+OPT_HAS_ARG['v']=0              OPT_VAL['v']=0
+
+OPT_NAME['E']='rate-diff'       OPT_DESC['E']="ALSA conformance --rate-criteria-diff-pct (difference, %)."
+OPT_HAS_ARG['E']=1              OPT_VAL['E']=''
+
+OPT_NAME['e']='rate-err'        OPT_DESC['e']="ALSA conformance --rate-err-criteria (max rate error)."
+OPT_HAS_ARG['e']=1              OPT_VAL['e']=''
+
+OPT_NAME['a']='avail-delay'     OPT_DESC['a']="ALSA conformance --avail-delay"
+OPT_HAS_ARG['a']=0              OPT_VAL['a']=0
+
+OPT_NAME['T']='test-suites'     OPT_DESC['T']="ALSA conformance --test-suites (Default: all)."
+OPT_HAS_ARG['T']=1              OPT_VAL['T']=''
+
+func_opt_parse_option "$@"
+
+alsa_device=${OPT_VAL['d']}
+pcm_p=${OPT_VAL['p']}
+pcm_c=${OPT_VAL['c']}
+rates=$([ -z "${OPT_VAL['r']}" ] && echo '' || echo "--allow-rates ${OPT_VAL['r']}")
+formats=$([ -z "${OPT_VAL['F']}" ] && echo '' || echo "--allow-formats ${OPT_VAL['F']}")
```

Comment:
The `echo ''` is not needed.
```bash
rates=$([ -z "${OPT_VAL['r']}" ] || echo "--allow-rates ${OPT_VAL['r']}")
```

Also, this repetition is calling for some new `prefix_if_not_empty()` and `prefix_if_1()` functions. I know they would be useful outside this new script.

```bash
formats=$(prefix_if_not_empty --allow-formats "${OPT_VAL['F']}")
```